test/fuzz/basic.ml

```diff
@@ -348,83 +359,101 @@ let memory_fill = const Memory_fill
 let memory_init = const (Memory_init (Raw 0))
 
 let memarg =
-  let* offset = range 10 in
-  let+ align = range 10 in
+  let* offset = int in
+  let+ align = int in
   { offset; align}
 
-let i32_load_i32 =
+let i32_load =
   let+ memarg in
   I_load (S32, memarg)
 
-let i64_load_i32 =
+let i64_load =
   let+ memarg in
   I_load (S64, memarg)
 
-let f32_load_i32 =
+let f32_load =
   let+ memarg in
   F_load (S32, memarg)
 
-let f64_load_i32 =
+let f64_load =
   let+ memarg in
   F_load (S64, memarg)
 
-let i32_load8_i32 =
+let i32_load8 =
   let* memarg in
   let+ sx in
   I_load8 (S32, sx, memarg)
 
-let i32_load16_i32 =
+let i32_load16 =
   let* memarg in
   let+ sx in
   I_load16 (S32, sx, memarg)
 
-let i64_load8_i32 =
+let i64_load8 =
   let* memarg in
   let+ sx in
   I_load8 (S64, sx, memarg)
 
-let i64_load16_i32 =
+let i64_load16 =
   let* memarg in
   let+ sx in
   I_load16 (S64, sx, memarg)
 
-let i64_load32_i32 =
+let i64_load32 =
   let* memarg in
   let+ sx in
   I64_load32 (sx, memarg)
 
-let i32_store_i32 =
+let i32_store =
   let+ memarg in
   I_store (S32, memarg)
 
-let i64_store_i32 =
+let i64_store =
   let+ memarg in
   I_store (S64, memarg)
 
-let f32_store_i32 =
+let f32_store =
   let+ memarg in
   F_store (S32, memarg)
 
-let f64_store_i32 =
+let f64_store =
   let+ memarg in
   F_store (S64, memarg)
 
-let i32_store8_i32 =
+let i32_store8 =
   let+ memarg in
   I_store8 (S32, memarg)
 
-let i64_store8_i32 =
+let i64_store8 =
   let+ memarg in
   I_store8 (S64, memarg)
 
-let i32_store16_i32 =
+let i32_store16 =
   let+ memarg in
   I_store16 (S32, memarg)
 
-let i64_store16_i32 =
+let i64_store16 =
   let+ memarg in
   I_store16 (S64, memarg)
 
-let i64_store32_i32 =
+let i64_store32 =
   let+ memarg in
   I64_store32 memarg
+
+let data_active =
+  let* ind = option indice in   (* is full indice gen OK ? bug offset label ? *)
+  let* vt = val_type in
+  let+ inst = const_of_val_type vt in   (* only i32 ? *)
```

Comment:
This is indeed always an `int32`. It's the offset at which it'll be in the memory. You should make sure that the length of the data + the offset is lower than the memory initial memory size (you'll have to store this in the environment).

Reply:
Memo: For now, adding this data/memory size constraint does not seem to be a "must have"

Reply:
Could you add a TODO with a link to https://github.com/OCamlPro/owi/pull/28#discussion_r1213333308 ?

Reply:
This has been fixed but not referenced here, sorry.